tests/internal/pin_test.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2016-2022, Pulumi Corporation.  All rights reserved.
+
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFixupGradleVersion(t *testing.T) {
```

Comment:
maybe https://github.com/ben-manes/gradle-versions-plugin could help with this instead of a custom tool, but I haven't checked, just a thought